### PR TITLE
Adding Design for Tags

### PR DESCRIPTION
**Pull request by beanh66**
Adding designs for a Tags Component. @serenamarie125 @jjlangholtz please take a look!

### CODE REVIEW

**Comment by jjlangholtz** [comment]
@beanh66 Looks great! I think the design allows us to achieve parity with the OpsUI tag editing capabilities. 👍

File: UX/ui-service/Common/Tags/design.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Tags - Common Component
+* Tags are used in multiple areas throughout the UI.
+* Tags are made up of two parts, a category and value. Some categories allow for more than one value to be associated.  
+* Tags can be applied to a single item or multiple items at one time depending on the use case.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: Is it this, or is that multiple values can be applied to a single category?

File: UX/ui-service/Common/Tags/design.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+* In all cases, the Tag selection dropdowns and the tag labels should be visually consistent.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: how about something like "In all cases, the tag selection mechanism should be visually consistent." ?

File: UX/ui-service/Common/Tags/design.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+* As tags are added, they should be placed at the bottom of the list. Once saved, the tags should be reordered into alphabetic order so upon returning to the tags area, users can easily scan the list.
+
+## Simple Tags
+  * ![Image of a Simple Tag.](img/SimpleTags-01.png)  
+
+1. **Selection:**
+  * A combobox should be used for the tag selection so users can rely on the dropdown or they can type to filter the list of options.
+  * Once a category has been chosen from the first dropdown, the second dropdown should be enabled.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: can we reword to stay away from dropdown?

File: UX/ui-service/Common/Tags/design.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+2. **Labels:**
+  * Once the user selects an option from the second dropdown, the tag label should be added below the selection boxes.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: "Once the user selects the value from the list, ..."

File: UX/ui-service/Common/Tags/design.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  * The entire label should have a dark boarder with the category left aligned.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: border (typo)

File: UX/ui-service/Common/Tags/design.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  * The value should be displayed to the right of the category, with a lighter boarder to distinguish it.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: border (typo)

File: UX/ui-service/Common/Tags/design.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  * Each value should have an x icon right aligned to allow for removing it.
+
+  * ![Image of Label with a Single Value.](img/SingleValue.png)  
+  * If the Tag only has one value associated with the category, clicking the x on that value will remove the entire tag.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: "If the tag category only has one associated value," seems easier to read

File: UX/ui-service/Common/Tags/design.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  * The value should be displayed to the right of the category, with a lighter boarder to distinguish it.
+  * Each value should have an x icon right aligned to allow for removing it.
+
+  * ![Image of Label with a Single Value.](img/SingleValue.png)  
+  * If the Tag only has one value associated with the category, clicking the x on that value will remove the entire tag.  
+
+3. **Multi-Value Tags:**   
+  * For cases where the category allows for more than one value to be associated, the second dropdown will indicate the values that have already been applied using the multiple select dropdown widget.
+  * Implementation Details: Use the [Bootstrap Multiple Select](http://www.patternfly.org/pattern-library/widgets/#bootstrap-select) for the values dropdown.
+  * Values can be removed by either deselecting items in the dropdown or by clicking the x next to the tag.
```
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: dropdown again L)

File: UX/ui-service/Common/Tags/design.md (hunk 1 above)
Comment (beanh66) [inline UX/ui-service/Common/Tags/design.md]: @serenamarie125 it is this. In some cases (Blueprints) the user is tagging a single item. In other cases, (Services) the user can multi select to tag multiple items at once. I address the multiple values per category later in the PR. Does that make sense?

File: UX/ui-service/Common/Tags/design.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+* Tags are made up of two parts, a category and value. Some categories allow for more than one value to be associated.  
+* Tags can be applied to a single item or multiple items at one time depending on the use case.
+* In all cases, the Tag selection dropdowns and the tag labels should be visually consistent.
+* As tags are added, they should be placed at the bottom of the list. Once saved, the tags should be reordered into alphabetic order so upon returning to the tags area, users can easily scan the list.
+
+## Simple Tags
+  * ![Image of a Simple Tag.](img/SimpleTags-01.png)  
+
+1. **Selection:**
+  * A combobox should be used for the tag selection so users can rely on the dropdown or they can type to filter the list of options.
```
Comment (beanh66) [inline UX/ui-service/Common/Tags/design.md]: @serenamarie125 I guess I don't see why we wouldn't use the BS Combobox. I didn't follow the + Search comment you made. The BS Combobox allows the user to type in order to filter the list down so I think that is what we need, right?

File: UX/ui-service/Common/Tags/design.md (hunk 1 above)
Comment (serenamarie125) [inline UX/ui-service/Common/Tags/design.md]: ok makes sense @beanh66

**Pull request merged**